docs/linting/typed-linting/Monorepos.md

```diff
@@ -76,7 +75,6 @@ module.exports = {
   ],
   parser: '@typescript-eslint/parser',
   parserOptions: {
-    tsconfigRootDir: __dirname,
```

Comment:
I'm in general a little -1 on including it because for a couple of reasons:

* It's extra config complexity that users will question
* Answering those questions takes up time from our already-more-than-ideal-length setup experience

Since the vast majority of projects do include a `tsconfig.json`, I think having auto-detection for it means it's not so useful anymore. In other words: I think that the misconfiguration case is now going to be so rare that it's not worth the extra config complexity.

Reply:
the thing that I worry about is if you do something like this in our repo:
```bash
$ cd tools
$ yarn eslint ./postinstall.ts
```

Then:
- `yarn` will resolve `eslint` to the right place
- `eslint` will resolve the file fine and will find the eslint config fine
- `node` will execute with `cwd = $PROJECT_ROOT/tools`
- your code will throw because it'll reach `cwd` without finding a `tsconfig`

The error will be somewhat informative (`` project was set to `true` but couldn't find any tsconfig.json relative to '$PROJECT_ROOT/tools/postinstall.ts' within '$PROJECT_ROOT/tools' ``), but this would definitely be a confusing experience for users, IMO.

But if they had specified `tsconfigRootDir` then the lint run would have "just worked" as we could have understood their project layout correctly.

It's the reason we added this option in the first place - it's very, very common for people to run their lint run from a directory that isn't the project root, which would ofc break things as we'd attempt to resolve `./tsconfig.json` based on CWD.

On the CLI sure it's easy to work around this - instead of `yarn eslint`, run `(cd .. && yarn eslint`) to set the CWD correctly. But there's another case that's not possible to work around - if someone has their editor root set to a subfolder of the workspace.

IMO the extra line of config is worth it to make things "just work" and get ahead of potential user issues.

Reply:
Got it - this is good info, thanks. I'll add them back in.